BizStream/brain-rot-erator
# Feature/BRAIN 9/10 Mock DB & Refactoring

- Clips are now saved to localStorage with metadata after they're processed in Python
- Added util functions to sort the clips based on their createdAt prop in localStorage
- Created a custom hook to check clips expiresAt prop every minute to see if any need to be deleted from temp and localStorage
- Created metadata for each clip while processing each one in Python
Refactored EVERYTHING:

- Moved all api calls to Python server to pythonService file
- Removed scheduler from Python (this is now being done on the Nextjs side using the metadata)
- Extracted many functions from process_video with and without ad fill, so I no longer have to update both at the same time

QA:

1. Click 'Attach a movie'
2. Give it a title and submit without adding ad fill (title's can be duplicated)
3. Dismiss the warning popup
4. Make sure the clips are displayed in descending order by createdAt time (this is a part of their display name)
5. Select a clip and download it to make sure that still works. Also, be sure to check your localStorage to make sure the clip is deleted from there and no longer displayed
6. Wait 10 seconds to make the clips are deleted from the screen and local (the timer is set to 10 seconds for testing at the moment)
7. Repeat steps 1-6 but provide ad fill. If you didn't have time to do all the checks (sorted correctly and can download/delete one) then perform the others this time around
